core/tests/mod.rs

```diff
@@ -0,0 +1,646 @@
+        }),
+    );
+
+    for block_height in 1..10 {
+        {
+            let mut success = false;
+            let mut last_block_sync = Instant::now();
+            for _ in 0..300 {
+                if last_block_sync.elapsed().as_secs() > 5 {
+                    let sumeragis = sumeragis.lock().unwrap();
+                    let mut all_blocks = Vec::new();
+                    for sumeragi in sumeragis.values() {
+                        sumeragi.apply_wsv(|wsv| all_blocks.extend(wsv.all_blocks_by_value()));
+                    }
+                    for sumeragi in sumeragis.values() {
+                        for block in &all_blocks {
+                            sumeragi.incoming_message(MessagePacket::new(
+                                ProofChain::default(),
+                                Message::BlockSyncUpdate(block.clone().into()),
+                            ));
+                        }
+                    }
+                    last_block_sync = Instant::now();
+                }
+
+                {
+                    let sumeragis = sumeragis.lock().unwrap();
+                    success = true;
+                    for sumeragi in sumeragis.values() {
+                        if sumeragi.apply_wsv(|wsv| {
+                            wsv.block_hashes_after_hash(None).len() != block_height
+                        }) {
+                            success = false;
+                        }
+                    }
+                    if success {
+                        break;
+                    }
+                }
+                sleep_ms(100);
+            }
+            assert!(success, "Not all sumeragi instances committed the block.");
+        }
+
+        let account_id: AccountId = "alice@wonderland".parse().unwrap();
+        let asset_definition_id: AssetDefinitionId = "rose#wonderland".parse().expect("Valid");
+        let mint_asset = MintBox::new(
+            1_u32.to_value(),
+            IdBox::AssetId(AssetId::new(
+                asset_definition_id.clone(),
+                account_id.clone(),
+            )),
+        );
+
+        let transaction =
+            TransactionBuilder::new(account_id.clone(), vec![mint_asset.into()], 100_000)
+                .sign(get_key_pair())
+                .unwrap();
+        let transaction = AcceptedTransaction::accept::<false>(
+            transaction,
+            &blank_config().wsv.transaction_limits,
+        )
+        .unwrap();
+
+        for peer_id in &peer_ids {
+            let sumeragis = sumeragis.lock().unwrap();
+            let wsv = sumeragis
+                .get(peer_id)
+                .unwrap()
+                .apply_wsv(WorldStateView::clone);
+            let queue = queues.get(peer_id).unwrap();
+            queue.push(transaction.clone().into(), &wsv).unwrap();
+        }
+    }
+}
+
+#[test]
+fn sumeragi_with_faults_commit_blocks() {
+    init_telemetry();
+
+    let peer_count = 7;
```

Comment:
Better to make it `const`

Reply:
It's more descriptive, but not better IMO. 

The compiler already knows that this is a value binding. Everything is immutable, and no enclosing scope is present, so it's effectively a compile-time constant.